src/ctap/data_formats.rs

```diff
@@ -17,12 +17,23 @@ use alloc::collections::BTreeMap;
 use alloc::string::String;
 use alloc::vec::Vec;
 use arrayref::array_ref;
-use cbor::{cbor_array_vec, cbor_bytes_lit, cbor_map_options, destructure_cbor_map};
+use cbor::{cbor_array_vec, cbor_map, cbor_map_options, destructure_cbor_map};
 use core::convert::TryFrom;
 use crypto::{ecdh, ecdsa};
 #[cfg(test)]
 use enum_iterator::IntoEnumIterator;
 
+// This is the algorithm specifier that is supposed to be used in a COSE key
+// map in ECDH. CTAP requests -25 which represents ECDH-ES + HKDF-256 here:
+// https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+const ECDH_ALGORITHM: i64 = -25;
+// This is the identifier used for ECDSA and ECDH in OpenSSH.
+const ES256_ALGORITHM: i64 = -7;
+// The COSE key parameter behind map key 1.
+const EC2_KEY_TYPE: i64 = 2;
+// The COSE key parameter behind map key -1.
+const P_256_CURVE: i64 = 1;
```

Comment:
All of these constants are specific to the COSE specification. It would be worth to either put them in the `CoseKey` namespace (i.e. `impl CoseKey { const FOO: i64 = bar }`, or prefix them with `COSE_`.

Reply:
I left `ES256_ALGORITHM` on top since it's used across in `SignatureAlgorithm` and `CoseKey`. All other constants are moved into `impl CoseKey`.